src/types.ts

```diff
@@ -0,0 +1,98 @@
+import Node from './node-internal'
+import RDFlibVariable from './variable'
+import RDFlibBlankNode from './blank-node'
+import Collection from './collection'
+import RDFlibLiteral from './literal'
+import RDFlibNamedNode from './named-node'
+import RDFlibDefaultGraph from './default-graph'
+import { DataFactory } from './factories/factory-types'
+import IndexedFormula from './store'
+import Fetcher from './fetcher'
+import Statement from './statement'
+import Empty from './empty'
+import { NamedNode, Term } from './tf-types'
+
+export const NamedNodeTermType = "NamedNode" as const
+export const BlankNodeTermType = "BlankNode" as const
+export const LiteralTermType = "Literal" as const
+export const VariableTermType = "Variable" as const
+export const DefaultGraphTermType = "DefaultGraph" as const
+// Non-RDF/JS types:
+export const CollectionTermType = "Collection" as const
+export const EmptyTermType = "Empty" as const
+export const GraphTermType = "Graph" as const
+
+export type TermType = typeof NamedNodeTermType
+  | typeof BlankNodeTermType
+  | typeof LiteralTermType
+  | typeof VariableTermType
+  | typeof DefaultGraphTermType
+  | typeof CollectionTermType
+  | typeof EmptyTermType
+  | typeof GraphTermType
+
+/**
+ * A valid mime type header
+ * @todo Convert these to const enums when it's supported https://github.com/babel/babel/issues/8741
+ */
+export enum ContentType {
+  rdfxml = "application/rdf+xml",
+  turtle = "text/turtle",
+  turtleLegacy = "application/x-turtle",
+  n3 = "text/n3",
+  n3Legacy = "application/n3",
+  nTriples = "application/n-triples",
+  nQuads = "application/n-quads",
+  nQuadsAlt = "application/nquads",
+  jsonld = "application/ld+json",
+  xhtml = "application/xhtml+xml",
+  html = "text/html",
+  sparqlupdate = "application/sparql-update",
+}
```

Comment:
I think this might work as a union type as well?

Reply:
Agreed, changed.